ansible/roles/test/files/ptftests/py3/generic_hash_test.py

```diff
@@ -379,8 +379,17 @@ def check_ip_route(self, pkt, masked_expected_pkt, sending_port):
         send the packet and check it is received by one of the expected ports
         """
         testutils.send_packet(self, sending_port, pkt)
-        port_index, received = testutils.verify_packet_any_port(
-            self, masked_expected_pkt, self.expected_port_list, timeout=0.1)
+        try:
+            port_index, received = testutils.verify_packet_any_port(
+                self, masked_expected_pkt, self.expected_port_list, timeout=0.1)
+        except AssertionError:
```

Comment:
Why not add the for loop at line 380 to make the logic easier? And will the retry cover some real issue?

Reply:
> Why not add the for loop at line 380 to make the logic easier? And will the retry cover some real issue?

It just a exception handling method when the PTF failed to send packet at our shared test server. No need to loop send packet. And as this is dedicate for packet hashing validation, packet retry would not affect the hash result. And we have many other cases for packet forwarding validation as well.